### ISSUE TITLE
[3.2.3] Tree sorting ignores resource id if called from icon above the tree

### ISSUE THREAD

**Issue by elcreator** [evolution-cms/evolution#2126]
Tree sorting ignores selected (or highest parent resource id available for logged in manager) resource id if called from icon above the tree but works if called from the context menu.
I.e. it always calls backend with id=0 if called from the icon but calls with the resource id if called from the context menu on this resource.
It leads to inability to sort for everyone except admin who has rights for resource with id = 0

**Comment by Seiger** [comment]
@elcreator Can You share screen

**Comment by elcreator** [comment]
![Screenshot 2023-08-29 192610](https://github.com/evolution-cms/evolution/assets/974975/5d1e455f-aff5-491c-9dc9-3e8d2a82fe9e)

**Comment by Seiger** [comment]
Please, check it with last updade core to version 3.2.3.

**Comment by elcreator** [comment]
no, it was not helpful. Looks like it was only merged to 3.2.3 without additional fixes (I already performed updating this file from June 16 commit before).

**Comment by elcreator** [comment]
just try to create non-admin manager user, assign to his group some resources, login with this profile and press on Sort button above the tree. Also here is my config regarding access to the root path:
![image](https://github.com/evolution-cms/evolution/assets/974975/f6ecb44c-1d1d-478c-bb93-a6c0c59e3d0e)

**Comment by middleDuckAi** [comment]
Thanks, reviewed this one together with the local PRD/SPEC package.

What looks most likely so far:
- The report already points to a very concrete manager-tree surface.
- This looks like the top sort icon building its request with a hardcoded `id=0` instead of the currently selected resource id.
- If that is correct, the fix may touch both the blade markup and the theme JS that consumes the generated URL.

Likely code areas to inspect:
- `manager/views/frame/tree.blade.php`
- `manager/media/style/default/js/modx.js`
- `manager/media/style/liquid/js/modx.js`

Current task status:
- ✅ PRD/SPEC checked.
- ✅ Enough information to start code analysis: yes.
- 🟡 Reproduced locally: not yet.
- ✅ Waiting on reporter: no.

**Comment by middleDuckAi** [comment]
Status update from MiddleDuck:

A fix is ready and the PR is open: #2235

What is included:
- replace the hardcoded top-tree sort URL with a contextual tree handler
- resolve the sort target from the current node, selected node, or first available tree node
- add a regression test that guards against `id=0` being reintroduced in the tree toolbar

Validation completed:
- `./vendor/bin/pest tests/Unit/TreeSortButtonContextTest.php`
- `php -l manager/views/frame/tree.blade.php`
- `php -l core/tests/Unit/TreeSortButtonContextTest.php`
- `node` syntax check for both manager theme `modx.js` files
